File: word-count/WordCount.fs
module WordCount
open System

let countWords (phrase:string) = 
    let filter (c:char) = System.Char.IsDigit c || System.Char.IsLetter c || c = '\'' || c = ' ' || c = ','
    let clean = phrase.ToLowerInvariant() |> Seq.filter filter |> System.String.Concat
    let words = clean.Split([|' ';','|], StringSplitOptions.RemoveEmptyEntries) |> Array.map (fun x -> x.Trim('\''))
    words |> Array.groupBy id |> Array.map (fun x -> (fst x, Array.length (snd x))) |> Map.ofArray

File: csharp/word-count/WordCount.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class WordCount
{
    private static readonly char[] splitChars = 
        new[] { ' ', ',', ':', '!', '&', '@', '$', '%', '^', '\n', '.' };

    public static IDictionary<string, int> CountWords(string phrase)
        => phrase.ToLower()
            .Split(splitChars, StringSplitOptions.RemoveEmptyEntries)
            .Select(w => w.Trim('\'', '"'))
            .GroupBy(o => o)
            .ToDictionary(o => o.Key, o => o.Count());
}